### PR TITLE
Add Vagrantfile for reproducible tests and development environment

### PR DESCRIPTION
**Pull request by michaeltlombardi**
This commit adds a vagrant box that pulls in the project
dependencies to make working on the code base easier.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  # config.vm.network "public_network"
+
+  # Share an additional folder to the guest VM. The first argument is
+  # the path on the host to the actual folder. The second argument is
+  # the path on the guest to mount the folder. And the optional third
+  # argument is a set of non-required options.
+  # config.vm.synced_folder "../data", "/vagrant_data"
+
+  config.vm.provider "virtualbox" do |vb|
+    vb.gui = true
```
Comment (DavidS) [inline Vagrantfile:49]:
```suggestion
    vb.gui = false
```
I prefer to use `vagrant ssh` to access the machine. @sanfrancrisko ?

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  config.vm.provision "shell", inline: <<-SHELL
+    $ChocolateyPackages = @(
+      'vscode'
+      'vscode-powershell'
+      'googlechrome'
+      'git'
+      'pester'
+      'pdk'
+    )
+    choco install $ChocolateyPackages -y
```
Comment (DavidS) [inline Vagrantfile:71]:
```suggestion
    choco install $ChocolateyPackages -y --no-progress
```

avoid pages and pages of download %%%

**Comment by DavidS** [comment]
Other than the two minor changes, works for me.

File: Vagrantfile (hunk 1 above)
Comment (sanfrancrisko) [reply Vagrantfile:49]: I do use RDP most of the time, but should probably use SSH, to be honest.

Comment (sanfrancrisko) [reply Vagrantfile:49]: Given VSCode is being pulled in, maybe we want to keep it enabled?

Comment (DavidS) [reply Vagrantfile:49]: Sure.

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  # Share an additional folder to the guest VM. The first argument is
+  # the path on the host to the actual folder. The second argument is
+  # the path on the guest to mount the folder. And the optional third
+  # argument is a set of non-required options.
+  # config.vm.synced_folder "../data", "/vagrant_data"
+
+  config.vm.provider "virtualbox" do |vb|
+    vb.gui = true
+    vb.memory = "4096"
```
Comment (DavidS) [inline Vagrantfile:50]:
```suggestion
    vb.cpus = 4
    vb.memory = "4096"
```

**Pull request merged**